# In-class Exercise: Folium Maps

In [1]:
import pandas as pd
import numpy as np
import folium
from folium.plugins import MarkerCluster, HeatMap

### Load some datasets

In [2]:
# meteorites: # meteorites: https://www.kaggle.com/nasa/meteorite-landings
df_me=pd.read_csv('lecture22_meteorites.csv')
df_me

name     id nametype              recclass      mass   fall  \
0          Aachen      1    Valid                    L5      21.0   Fell   
1          Aarhus      2    Valid                    H6     720.0   Fell   
2            Abee      6    Valid                   EH4  107000.0   Fell   
3        Acapulco     10    Valid           Acapulcoite    1914.0   Fell   
4         Achiras    370    Valid                    L6     780.0   Fell   
...           ...    ...      ...                   ...       ...    ...   
45711  Zillah 002  31356    Valid               Eucrite     172.0  Found   
45712      Zinder  30409    Valid  Pallasite, ungrouped      46.0  Found   
45713        Zlin  30410    Valid                    H4       3.3  Found   
45714   Zubkovsky  31357    Valid                    L6    2167.0  Found   
45715  Zulu Queen  30414    Valid                  L3.7     200.0  Found   

         year    reclat    reclong               GeoLocation  
0      1880.0  50.77500    6.08333     (50.775000, 6.083330)  
1      1951.0  56.18333   10.23333    (56.183330, 10.233330)  
2      1952.0  54.21667 -113.00000  (54.216670, -113.000000)  
3      1976.0  16.88333  -99.90000   (16.883330, -99.900000)  
4      1902.0 -33.16667  -64.95000  (-33.166670, -64.950000)  
...       ...       ...        ...                       ...  
45711  1990.0  29.03700   17.01850    (29.037000, 17.018500)  
45712  1999.0  13.78333    8.96667     (13.783330, 8.966670)  
45713  1939.0  49.25000   17.66667    (49.250000, 17.666670)  
45714  2003.0  49.78917   41.50460    (49.789170, 41.504600)  
45715  1976.0  33.98333 -115.68333  (33.983330, -115.683330)  

[45716 rows x 10 columns]

In [3]:
# global volcanos
df_volc = pd.read_csv("lecture20_volc.csv")
df_volc

Number                       Name         Country  \
0     210010  West Eifel Volcanic Field         Germany   
1     210020            Chaine des Puys          France   
2     210030        Olot Volcanic Field           Spain   
3     210040   Calatrava Volcanic Field           Spain   
4     211001                 Larderello           Italy   
...      ...                        ...             ...   
1503  390130                 Zavodovski  United Kingdom   
1504  390140        Protector Seamounts  United Kingdom   
1505  390812            Rittmann, Mount      Antarctica   
1506  390829          James Ross Island      Antarctica   
1507  390847                   Melville      Antarctica   

                              Region                 Type  \
0     Mediterranean and Western Asia              Maar(s)   
1     Mediterranean and Western Asia         Lava dome(s)   
2     Mediterranean and Western Asia  Pyroclastic cone(s)   
3     Mediterranean and Western Asia  Pyroclastic cone(s)   
4     Mediterranean and Western Asia  Explosion crater(s)   
...                              ...                  ...   
1503                      Antarctica        Stratovolcano   
1504                      Antarctica            Submarine   
1505                      Antarctica               Shield   
1506                      Antarctica               Shield   
1507                      Antarctica        Stratovolcano   

         Activity Evidence Last Known Eruption  Latitude  Longitude  \
0           Eruption Dated            8300 BCE    50.170      6.850   
1           Eruption Dated            4040 BCE    45.775      2.970   
2        Evidence Credible             Unknown    42.170      2.530   
3           Eruption Dated            3600 BCE    38.870     -4.020   
4        Eruption Observed             1282 CE    43.250     10.870   
...                    ...                 ...       ...        ...   
1503     Eruption Observed             2016 CE   -56.300    -27.570   
1504     Eruption Observed             1962 CE   -55.912    -28.167   
1505  Unrest / Pleistocene             Unknown   -73.450    165.500   
1506     Evidence Credible             Unknown   -64.150    -57.750   
1507    Evidence Uncertain             Unknown   -62.020    -57.670   

      Elevation (Meters)                Dominant Rock Type  \
0                    600                           Foidite   
1                   1464             Basalt / Picro-Basalt   
2                    893  Trachybasalt / Tephrite Basanite   
3                   1117             Basalt / Picro-Basalt   
4                    500                           No Data   
...                  ...                               ...   
1503                 551             Basalt / Picro-Basalt   
1504                 -55                          Rhyolite   
1505                2600                               NaN   
1506                1630             Basalt / Picro-Basalt   
1507                 549                               NaN   

                                  Tectonic Setting  
0           Rift Zone / Continental Crust (>25 km)  
1           Rift Zone / Continental Crust (>25 km)  
2          Intraplate / Continental Crust (>25 km)  
3          Intraplate / Continental Crust (>25 km)  
4     Subduction Zone / Continental Crust (>25 km)  
...                                            ...  
1503     Subduction Zone / Oceanic Crust (< 15 km)  
1504     Subduction Zone / Oceanic Crust (< 15 km)  
1505       Intraplate / Continental Crust (>25 km)  
1506       Intraplate / Continental Crust (>25 km)  
1507       Intraplate / Continental Crust (>25 km)  

[1508 rows x 12 columns]

In [4]:
# global earthquakes
df_eq = pd.read_csv("lecture20_eq.csv")
df_eq

Date      Time  Latitude  Longitude        Type   Depth  \
0      01/02/1965  13:44:18   19.2460   145.6160  Earthquake  131.60   
1      01/04/1965  11:29:49    1.8630   127.3520  Earthquake   80.00   
2      01/05/1965  18:05:58  -20.5790  -173.9720  Earthquake   20.00   
3      01/08/1965  18:49:43  -59.0760   -23.5570  Earthquake   15.00   
4      01/09/1965  13:32:50   11.9380   126.4270  Earthquake   15.00   
...           ...       ...       ...        ...         ...     ...   
23407  12/28/2016  08:22:12   38.3917  -118.8941  Earthquake   12.30   
23408  12/28/2016  09:13:47   38.3777  -118.8957  Earthquake    8.80   
23409  12/28/2016  12:38:51   36.9179   140.4262  Earthquake   10.00   
23410  12/29/2016  22:30:19   -9.0283   118.6639  Earthquake   79.00   
23411  12/30/2016  20:08:28   37.3973   141.4103  Earthquake   11.94   

       Depth Error  Depth Seismic Stations  Magnitude Magnitude Type  ...  \
0              NaN                     NaN        6.0             MW  ...   
1              NaN                     NaN        5.8             MW  ...   
2              NaN                     NaN        6.2             MW  ...   
3              NaN                     NaN        5.8             MW  ...   
4              NaN                     NaN        5.8             MW  ...   
...            ...                     ...        ...            ...  ...   
23407          1.2                    40.0        5.6             ML  ...   
23408          2.0                    33.0        5.5             ML  ...   
23409          1.8                     NaN        5.9            MWW  ...   
23410          1.8                     NaN        6.3            MWW  ...   
23411          2.2                     NaN        5.5             MB  ...   

       Magnitude Seismic Stations  Azimuthal Gap  Horizontal Distance  \
0                             NaN            NaN                  NaN   
1                             NaN            NaN                  NaN   
2                             NaN            NaN                  NaN   
3                             NaN            NaN                  NaN   
4                             NaN            NaN                  NaN   
...                           ...            ...                  ...   
23407                        18.0          42.47                0.120   
23408                        18.0          48.58                0.129   
23409                         NaN          91.00                0.992   
23410                         NaN          26.00                3.553   
23411                       428.0          97.00                0.681   

       Horizontal Error  Root Mean Square            ID  Source  \
0                   NaN               NaN  ISCGEM860706  ISCGEM   
1                   NaN               NaN  ISCGEM860737  ISCGEM   
2                   NaN               NaN  ISCGEM860762  ISCGEM   
3                   NaN               NaN  ISCGEM860856  ISCGEM   
4                   NaN               NaN  ISCGEM860890  ISCGEM   
...                 ...               ...           ...     ...   
23407               NaN            0.1898    NN00570710      NN   
23408               NaN            0.2187    NN00570744      NN   
23409               4.8            1.5200    US10007NAF      US   
23410               6.0            1.4300    US10007NL0      US   
23411               4.5            0.9100    US10007NTD      US   

      Location Source Magnitude Source     Status  
0              ISCGEM           ISCGEM  Automatic  
1              ISCGEM           ISCGEM  Automatic  
2              ISCGEM           ISCGEM  Automatic  
3              ISCGEM           ISCGEM  Automatic  
4              ISCGEM           ISCGEM  Automatic  
...               ...              ...        ...  
23407              NN               NN   Reviewed  
23408              NN               NN   Reviewed  
23409              US               US   Reviewed  
23410              US

### Q1. Extract the latitude and longitude values from the meteorites dataframe, removing NaN values 

### Q2. Use `folium` to plot the Q1 (lat,lon) values with circle markers and the color of your choice. 

This a global map, so choose and appropriate `location` and `zoom_start`.

### Q3. Now make a `folium` heatmap of the meteorite lat/lon values

### Q4. Switching datasets now... Extract the latitude and longitude values for both volcanoes and earthquakes between latitudes 45 to 53 and longitudes -125 to -120.

### Q5. Use `folium` to plot the volcano (lat,lon) values with red triangles and earthquake (lat,lon) values with blue circles. Select an appropriate map location near the center of the study region, and use a `zoom_start=5.5`. 

### Save your notebook and output an html version. To receive credit for this in-class exercise upload both the .html and .ipynb file.